File: 新・Bランクレベルアップメニュー/マップの扱い/four.js
const main = (input) => {
    const [h, w, d] = input.trim().split(' ').map(Number);
    const nums = Array(h).fill().map( () => Array(w).fill(0));
    let count = 1;
    const diagonal = (num1, num2) => {
        let y = num1, x = num2;
        while (y >= 0 && x < w) {
            nums[y][x] = count++;
            y--;
            x++;
        }
        
    }
    
    const rDiagonal = (num1, num2) => {
        let y = num1, x = num2;
        while (x >= 0 && y < h) {
            nums[y][x] = count++;
            y++;
            x--;
        }
        
    }
    if (d === 1) {
        for (let i = 0; i < h; i++) {
            diagonal(i, 0);
        }
        for (let i = 1; i < w; i++) {
            diagonal(h - 1, i)
        }
    } else if (d === 2) {
        for (let i = 0; i < h; i++) {
            for (let j = 0; j < w; j++) {
                nums[i][j] = count++;
            }
        }
    } else if (d === 3) {
        for (let i = 0; i < w; i++) {
            for (let j = 0; j < h; j++) {
                nums[j][i] = count++;
            }
        }
    } else if (d === 4) {
        for (let i = 0; i < w; i++) {
            rDiagonal(0, i);
        }
        for (let i = 1; i < h; i++) {
            rDiagonal(i, w - 1);
        }
    }
    console.log(nums.map(num => num.join(' ')).join('\n'));
}
main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: 新・Bランクレベルアップメニュー/マップの扱い/four.rb
h, w, d = gets.split.map(&:to_i)
nums = Array.new(h) { Array.new(w, 0) }
count = 1
  
diagonal = lambda do |y, x|
    while y >= 0 && x < w
      nums[y][x] = count
      count += 1
      y -= 1
      x += 1
    end
end
  
r_diagonal = lambda do |y, x|
    while x >= 0 && y < h
      nums[y][x] = count
      count += 1
      y += 1
      x -= 1
    end
end
  
case d
when 1
    (0...h).each do |i|
      diagonal.call(i, 0)
    end
    (1...w).each do |i|
      diagonal.call(h - 1, i)
    end
when 2
    (0...h).each do |i|
      (0...w).each do |j|
        nums[i][j] = count
        count += 1
      end
    end
when 3
    (0...w).each do |i|
      (0...h).each do |j|
        nums[j][i] = count
        count += 1
      end
    end
when 4
    (0...w).each do |i|
      r_diagonal.call(0, i)
    end
    (1...h).each do |i|
      r_diagonal.call(i, w - 1)
    end
end

puts nums.map { |row| row.join(' ') }